sycl/doc/extensions/IntelGPU/IntelGPUDeviceInfo.md

```diff
@@ -9,9 +9,50 @@ This proposal details what is required to provide this information as a SYCL ext
 
 ## Feature Test Macro ##
 
-The Feature Test Macro will be defined as:
+The Feature Test Macro SYCL\_EXT\_INTEL\_DEVICE\_INFO will be defined as one of the values defined in the table below. The existence of this macro can be tested to determine if the implementation supports this feature, or applications can test the macro's value to determine which of the extension's APIs the implementation supports.
+
+| Value | Description |
+| ----- | ----------- |
+| 1     | Initial extension version\. Base features are supported |
+| 2     | Device UUID is supported |
+
+
+# Device UUID #
+
+A new device descriptor will be added which will provide the device Universal Unique ID (UUID).
+
+This new device descriptor is currently only available for devices in the Level Zero platform, and the matching aspect is only true for those devices. The DPC++ default behavior would be to expose the UUIDs of all supported GPU devices which enables detection of total number of unique devices.
+
+
+## Version ##
+
+The extension supports this query in version 2 and later.
+
+
+## Device Information Descriptors ##
+
+| Device Descriptors | Return Type | Description |
+| ------------------ | ----------- | ----------- |
+| info\:\:device\:\:ext\_intel\_device\_info\_uuid | uint8\_t \* | Returns the device UUID|
```

Comment:
Why did you change the return type here?  Returning `std::array<std::byte, 16>` was much better, I think.  With `uint8_t *`, it's unclear whether the caller is supposed to deallocate the memory.

Reply:
@smaslov-intel pointed out that it's preferable to not use STLs to maintain C compatibility. Changing this does include the requirement for deallocation.

Reply:
my bad, will fix it.